src/composables/map/street-view.composable.ts

```diff
@@ -0,0 +1,257 @@
+          updatePanorama
+        )
+      }
+    }
+  }
+
+  function updatePanorama(data: any, status: any) {
+    if (status === googleService?.maps.StreetViewStatus.OK) {
+      noDataAtLocation.value = false
+      nextTick(() => {
+        panorama.setPosition(data.location.latLng)
+        panorama.setVisible(true)
+      })
+    } else {
+      noDataAtLocation.value = true
+      svFeature.value = undefined
+      panorama.setVisible(false)
+    }
+  }
+
+  watch(infoOpen, open => {
+    if (open) {
+      if (isStreetviewActive.value && locationInfo.value) {
+        setSvFeatures(locationInfo.value)
+      }
+    } else {
+      svFeature.value = undefined
+    }
+  })
+
+  watch(locationInfo, loc => {
+    if (loc) {
+      if (isStreetviewActive.value && !panoPositionChanging.value) {
+        setLocation(loc)
+      }
+    } else {
+      svFeature.value = undefined
+      map.getViewport().style.cursor = ''
+    }
+  })
+
+  watch(isStreetviewActive, async act => {
+    if (act) {
+      await loadGoogleapis()
+      if (window.hasOwnProperty('google')) {
+        // @ts-ignore
+        googleService = window.google
+      }
+      // todo PIWIK
+      if (streetViewService === null) {
+        streetViewService = new googleService!.maps.StreetViewService()
+      }
+      if (panorama === null) {
+        panorama = new googleService!.maps.StreetViewPanorama(
+          document.getElementById('streetview-div'),
+          {
+            pov: {
+              heading: 0,
+              pitch: 0,
+              zoom: 1,
+            },
+            visible: false,
+            zoom: 1,
+          }
+        )
+      }
+      if (panoramaLinksListener === null) {
+        panoramaLinksListener = googleService?.maps.event.addListener(
+          panorama,
+          'links_changed',
+          () => handlePanoramaPositionChange(true)
+        )
+      }
+      if (panoramaPovListener === null) {
+        panoramaPovListener = googleService?.maps.event.addListener(
+          panorama,
+          'pov_changed',
+          () => handlePanoramaPositionChange(false)
+        )
+      }
+      setLocation(locationInfo.value)
+    } else {
+      if (panorama !== null) {
+        panorama.setVisible(false)
+      }
+      svFeature.value = undefined
+      if (panoramaPovListener) {
+        googleService?.maps.event.removeListener(panoramaPovListener)
+        panoramaPovListener = null
+      }
+      if (panoramaLinksListener) {
+        googleService?.maps.event.removeListener(panoramaLinksListener)
+        panoramaLinksListener = null
+      }
+    }
+  })
+
+  watch(svFeature, svf => {
+    svFeatureLayer.getSource()?.clear()
+    if (svf) {
+      svFeatureLayer.getSource()?.addFeature(svf.compass)
+      svf.directions.forEach((f: SvDirectionFeature) =>
+        svFeatureLayer.getSource()?.addFeature(f)
+      )
+      hidePointer.value = true
+      map.addInteraction(selectFeature)
+      map.on('pointermove', handleHover)
```

Comment:
This one `map.on('pointermove', handleHover)` can be moved outside if else as you doing it both times

Reply:
map.on != map.un